EIPS/eip-223.md

```diff
@@ -0,0 +1,385 @@
+created: 2017-05-03
+---
+
+## Abstract
+
+The following describes an interface for fungible tokens that supports a `tokenReceived` callback to notify contract recipients when tokens are received.
+
+## Motivation
+
+This token introduces a communication model for contracts that can be utilized to straighten the behavior of contracts that interact with such tokens. Specifically, this proposal:
+
+1. Informs receiving contracts of tokens, as opposed to [ERC-20](./eip-20.md) where the recipient of a token transfer gets no notification.
+2. Is more gas-efficient when depositing tokens to contracts.
+3. Allows for `_data` recording for financial transfers.
+
+
+## Specification
+
+### Methods
+
+Contracts intending to receive these tokens MUST implement `tokenReceived`.
+
+Token transfers to contracts not implementing `tokenReceived` as described below MUST revert.
+
+### Token contract
+
+#### `totalSupply`
+
+```solidity
+function totalSupply() view returns (uint256 totalSupply)
+```
+
+Get the total supply of the token. The functionality of the method is similar to `totalSupply()` of ERC-20. 
+
+#### `name`
+
+```solidity
+function name() view returns (string _name)
+```
+
+Get the name of the token. The functionality of the method is similar to `name()` of ERC-20.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+#### `symbol`
+
+```solidity
+function symbol() view returns (bytes32 _symbol)
+```
+
+Get the symbol of token. The functionality of the method is similar to `symbol()` of ERC-20.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+#### `decimals`
+
+```solidity
+function decimals() view returns (uint8 _decimals)
+```
+
+Get decimals of token. The functionality of the method is similar to `decimals()` of ERC-20.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+#### `balanceOf`
+
+```solidity
+function balanceOf(address _owner) view returns (uint256 balance)
+```
+
+Get the account balance of another account with address `_owner`. The functionality of the method is similar to `balanceOf(address)` of ERC-20.
+
+
+#### `transfer(address, uint)`
+
+```solidity
+function transfer(address _to, uint _value) returns (bool)
+```
+
+This function must transfer tokens and invoke the function `tokenReceived(address, uint256, bytes calldata)` in `_to`, if `_to` is a contract. If the `tokenReceived` function is not implemented in `_to` (recipient contract), then the transaction must fail and the transfer of tokens must be reverted.
+If `_to` is an externally owned address, then the transaction must be sent without executing `tokenReceived` in `_to`.
+ `_data` can be attached to this token transaction and it will stay in blockchain forever (requires more gas). `_data` can be empty.
```

Comment:
```suggestion
 `_data` can be attached to this token transaction, but it requires more gas. `_data` can be empty.
```
'Forever' will no longer be true if EIP-4444 is implemented.

Reply:
You are right.